API/Tensorflow/native_libs/src/graph/GraphSession.h

```diff
@@ -52,15 +48,31 @@ class GraphSession
 	}
 
 	template<TF_DataType DataTypeLabel>
-	Tensor<DataTypeLabel>** eval() const
+	Tensor<DataTypeLabel>** eval(const std::map<std::string, std::shared_ptr<Tensor<DataTypeLabel>>>& substitutions) const
```

Comment:
This is ok for now, but I think in the future we may want to also have an `eval :: List Placeholder -> List Tensor -> List Tensor` that doesn't need to use placeholder names (just using their pointers is the suggested way by TF C API).
For example, lift may not have to use the strings but just pointer values and get rid of using the map.

Reply:
Yes, it is doable.